src/main.rs

```diff
@@ -113,6 +113,20 @@ fn main() {
         _ => atty::is(Stream::Stdout),
     };
 
+    let path_separator: Option<u8> = match matches.value_of("path-separator") {
+        Some(sep_str) => {
+            let sep = sep_str.as_bytes();
+            if sep.len() != 1 {
+                print_error_and_exit!(
+                    "'{}' is not a valid path separator. See 'fd --help'.",
```

Comment:
Instead of referring to the help text, maybe just say explicitly what the problem is: *"The argument to --path-separator must be a single character"*.